### PR TITLE
Retrier v0.0.1

### PR DESCRIPTION
**Pull request by evgeni-spasov-metrilo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,15 @@
-Retrier
+`Retrier` is a small ruby class that helps you try to execute a block of code more than once.
```
Comment (velichkoStoev) [inline README.md]: Ruby*, capitalized ! :D

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+source 'https://rubygems.org'
+
+# Specify your gem's dependencies in better_env.gemspec
+gemspec
+
+group :development, :test do
+  gem 'pry-byebug'
+  gem 'rubocop'
+  gem 'fasterer'
```
Comment (velichkoStoev) [inline Gemfile:8]: What is this one for?

File: README.md [hunk 3]
```diff
@@ -1 +1,15 @@
...
+It catches a specific exception, waits certain amount of time and tries the code again until the retry count is exhausted.
```
Comment (velichkoStoev) [inline README.md]: waits for a* certain amount of time

File: retrier.gemspec [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+require 'retrier'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'retrier'
+  spec.version       = Retrier::VERSION
+  spec.authors       = ['Metrilo']
+  spec.email         = ['support@metrilo.com']
+
+  spec.summary       = 'Retry a ruby block.'
+  spec.description   = 'Retry a ruby block.'
```
Comment (velichkoStoev) [inline retrier.gemspec]: ```Retry a Ruby block for a specific error``` or smth sounds a bit better IMO.

**Review by velichkoStoev — CHANGES_REQUESTED** [review]

**Comment by velichkoStoev** [comment]
Great clean-up!

**Review by adamof — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+.DS_Store
```
Comment (yassenb) [inline .gitignore]: Don't put OS / user specific ignores here, only project related ones. Use `~/.gitignore` for globals such as this

File: .rubocop.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+AllCops:
```
Comment (yassenb) [inline .rubocop.yml:1]: A note for the future: we should pull the Rubocop (and other linters) configuration in a separate repo

File: Gemfile [hunk 7]
```diff
@@ -0,0 +1,11 @@
+source 'https://rubygems.org'
+
+# Specify your gem's dependencies in better_env.gemspec
```
Comment (yassenb) [inline Gemfile]: better_env?

File: README.md [hunk 8]
```diff
@@ -1 +1,15 @@
...
+
+You just pass: `ErrorClass`, `waiting_time` (seconds), `attempts`:
+```ruby
+retrier = Retrier.new(Mailgun::Error, waiting_time: 5, attempts: 3)
+
+retrier.execute do
+  # some request to Mailgun or other third-party API
```
Comment (yassenb) [inline README.md]: This can be any code, not just requests

File: retrier.gemspec [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  spec.homepage      = 'https://github.com/Metrilo/retrier'
+
+  spec.files         = `git ls-files -z`.split("\x0").reject do |f|
```
Comment (yassenb) [inline retrier.gemspec]: Quite hacky IMO

File: .ruby-version [hunk 10]
```diff
@@ -0,0 +1 @@
+2.3.1
```
Comment (yassenb) [inline .ruby-version]: Why no 2.5.0?

File: retrier.gemspec [hunk 11]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'retrier'
```
Comment (yassenb) [inline retrier.gemspec]: `require_relative` instead?

File: spec/spec_helper.rb [hunk 12]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+$LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
+require 'retrier'
```
Comment (yassenb) [inline spec/spec_helper.rb]: `require_relative`

File: spec/retrier_spec.rb [hunk 13]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Error < StandardError; end
```
Comment (yassenb) [inline spec/retrier_spec.rb]: Defining the class in the global namespace will leave it defined for other tests as well which might lead to confusion

File: spec/retrier_spec.rb [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+      expect do
+        subject.execute do
+          block_executions += 1
+
+          raise(Error, 'There is something wrong here')
+        end
+      end.to raise_error(Error, 'There is something wrong here')
+
+      expect(block_executions).to eq(attempts)
+    end
```
Comment (yassenb) [inline spec/retrier_spec.rb]: Add a test that it immediately (doesn't retry) raises the error if it's not of the specified class

**Review by yassenb — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 2 above)
Comment (evgeni-spasov-metrilo) [reply Gemfile:8]: Just a small tool for developers: https://github.com/DamirSvrtan/fasterer

**Review by velichkoStoev — APPROVED** [review]

**Review by yassenb — APPROVED** [review]

File: spec/retrier_spec.rb [hunk 15]
```diff
@@ -1,19 +1,22 @@
 # frozen_string_literal: true
 
-class Error < StandardError; end
+module RetrierTest
+  class RetriableError < StandardError; end
```
Comment (yassenb) [inline spec/retrier_spec.rb]: You can define those in the describe block, no need for a separate module

**Review by yassenb — APPROVED** [review]

**Pull request merged**